politeiawww/invoices.go

```diff
@@ -1027,6 +1022,35 @@ func (p *politeiawww) processEditInvoice(ei cms.EditInvoice, u *user.User) (*cms
 			ErrorCode: cms.ErrorStatusInvalidInvoiceEditMonthYear,
 		}
 	}
+
+	// Dupe address check.
+	invInput, err := parseInvoiceInput(ei.Files)
+	if err != nil {
+		return nil, err
+	}
+
+	invoiceAddress, err := p.cmsDB.InvoicesByAddress(invInput.PaymentAddress)
+	if err != nil {
+		return nil, www.UserError{
+			ErrorCode: cms.ErrorStatusInvalidPaymentAddress,
+		}
+	}
+	// If more than 1 invoice comes back with the same address, it's
+	// surely a duplicate.
+	if len(invoiceAddress) > 1 {
+		return nil, www.UserError{
+			ErrorCode: cms.ErrorStatusDuplicatePaymentAddress,
+		}
+	} else if len(invoiceAddress) == 1 &&
+		invInput.PaymentAddress != invRec.Input.PaymentAddress {
```

Comment:
This was kinda confusing to me.  Another way you might do this would be to replace these conditionals with something like the following.  Up to you.

```
	for _, v := range invoiceAddress {
		if v.Token != ei.Token {
			return nil, www.UserError{
				ErrorCode: cms.ErrorStatusDuplicatePaymentAddress,
			}
		}
	}
```